### PR TITLE
completed

### PR DESCRIPTION
**Pull request by DavonKenn**
# Submitting Week 5 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [x ] JavaScript Coding Challenges
- [x] Ruby Coding Challenges
- [x] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 // ASSESSMENT 5: JavaScript Coding Practical Questions with Jest
 
+// const { array } = require("yargs")
```
Comment (SunkissedQueen) [inline code-challenges.test.js:3]: Sneaky yargs

File: code-challenges.test.js [hunk 2]
```diff
@@ -10,47 +12,225 @@
 // Reminder: The test will call your function
 // Run the file with the following command: $ yarn jest
 
+// Boiler plate for my test!!
```
Comment (SunkissedQueen) [inline code-challenges.test.js:15]: love it

File: code-challenges.test.js [hunk 3]
```diff
@@ -10,47 +12,225 @@
...
 // Expected output: "G0bbl3dyg00k"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
-
+it("takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", () => {
+    expect(codeM(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+    expect(codeM(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+    expect(codeM(secretCodeWord3)).toEqual("3cc3ntr1c")
+  })
+})
```
Comment (SunkissedQueen) [inline code-challenges.test.js:42]: good

File: code-challenges.test.js [hunk 4]
```diff
@@ -10,47 +12,225 @@
...
+// codeM
+//   ✕ takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0. (7 ms)
+
+// ● codeM › takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
+
+//   expect(received).toEqual(expected) // deep equality
+
+//   Expected: "L4ck4d41s1c4l"
+//   Received: "Lackadaisical"
+//  I used a conditional statement with . replace to swtch th letter to a number it came back the same. What i did wrong was probally hard coding and putting it in a conditional statement.
```
Comment (SunkissedQueen) [inline code-challenges.test.js:78]: Does the `replace` method iterates across a string without the for loop?

File: code-challenges.test.js [hunk 5]
```diff
@@ -10,47 +12,225 @@
...
-// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
-
-// a) Create a test with expect statements using the variable provided.
-
+it("takes in an array of words and a single letter and returns all the words that contain that particular letter.", () => {
+    expect(partL(arrayOfWords1,letterA)).toEqual("Apple", "Banana", "Orange")
+    expect(partL(arrayOfWords2,letterE)).toEqual("Cherry", "Blueberry", "Peach")
+  })
+})
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:96]: Is there a reason that you changed the datatype of your expected output to be strings vice an array of strings?

File: code-challenges.test.js [hunk 6]
```diff
@@ -10,47 +12,225 @@
...
+// make a conditional statement
+// 
+// Taking in a array of words,
+// taking out words with  particular letter that was given.
+  
+const partL = (arr,val) => {  
+    let filteredVal = [];
+    for (let i = arr.length - 1; i >= 0; i--) {
+        if (val = (arr[i])) {
+            filteredVal.push(arr.splice(i, 1)[0]);
```
Comment (SunkissedQueen) [inline code-challenges.test.js:111]: Interesting syntax right here. Could you explain this in your pseudocode more?

File: code-challenges.test.js [hunk 7]
```diff
@@ -10,47 +12,225 @@
...
+//  Psuedo Code
+// declared a function called fulH
+// ill identify what is a full house first.
+// identify what is one pair and one three of a kind.
+// i would use an conditional staetment
+// create a variable for fullHouse
+//input:array of 5 number
+// output: true or false
+
+const fulHs = (arr) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:197]: The pseudocode has a good approach. I am not seeing that relfected in the function. 
`ill identify what is a full house first. identify what is one pair and one three of a kind.`
How would you explain a full house, one pair, and three of a kind? in plain words.

File: interview-questions.md [hunk 8]
```diff
@@ -7,53 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+A ruby gem is package library for ruby.Another name other for gems are plug ins and modules.Gems basically are ruby code in a huge file to make other life easier. People can create a gem and upload it to ruby gem server for everyone to use. Things that you can do with gem is add a frame work for testing like Rspect a gem  we used in class recently to test our ruby applications. You can upload photo using a gem called carrierwave and stripe is another gem used to take payments.Gems honestly make ruby alot more easier.
 
 
 3. What is Ruby on Rails?
 
-  Your answer:
+  Your answer: Ruby on rails is a gem in ruby.
 
-  Researched answer:
+  Researched answer: Ruby on rails is designed to make developers life easier by picking up what you need to get started on web applications.Ruby on rails wwas created by david hansson in 2003.Ruby on rails was a byproduct of basecamp. So with ruby on rails you wont have to write every single piece of code in a program as a developer you can refer to a framework,find a code for the common functions and you can plug it into your website or app.I recently used ruby on rails in class to create a database with my favorite movies in the challenges.
```
Comment (SunkissedQueen) [inline interview-questions.md:28]: interesting

File: interview-questions.md [hunk 9]
```diff
@@ -7,53 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: Relational database is something similiar to excel.It has a row of data and information about the data in the row.
 
   Researched answer:
-
+Relational database map out your data points and bring out the relations between them.So it basically stored data organized by columns tables and rows. I used this in class on pg admin and SQL finding data on life expectancy in diffferent countries.Other kinds of databases are NoSQL database,Operational database,Centralised database and Distributed database.
```
Comment (SunkissedQueen) [inline interview-questions.md:37]: nice

File: interview-questions.md [hunk 10]
```diff
@@ -7,53 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer:A primary key is a primary keyword in a table.
 
   Researched answer:
-
+A primary key is a key:value pair in a relational database that is id and a number.It is  Its important because an id and number are there so when you know to find the record in the database. It basically is like an address you can find a record knowing the id , number and the unique key.Primary keys were used in class to create a movies list of favorite movie and movie length. Ids and number was given to know which one was what and what movie length was what.
```
Comment (SunkissedQueen) [inline interview-questions.md:45]: great

File: interview-questions.md [hunk 11]
```diff
@@ -7,53 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 2. What is a gem?
 
-  Your answer:
+  Your answer: A gem is a tool used in ruby to help run test for ruby.
 
   Researched answer:
-
+A ruby gem is package library for ruby.Another name other for gems are plug ins and modules.Gems basically are ruby code in a huge file to make other life easier. People can create a gem and upload it to ruby gem server for everyone to use. Things that you can do with gem is add a frame work for testing like Rspect a gem  we used in class recently to test our ruby applications. You can upload photo using a gem called carrierwave and stripe is another gem used to take payments.Gems honestly make ruby alot more easier.
```
Comment (SunkissedQueen) [inline interview-questions.md:21]: awesome

File: interview-questions.md [hunk 12]
```diff
@@ -7,53 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a hash in Ruby?
 
-  Your answer:
+  Your answer: A hash in ruby is a storage for key value pairs. The curly braces hold key value pairs.In class I remember using hashes in the texted based game .
 
   Researched answer:
-
+So I was correct.A hash in ruby is a storage for key value pairs. The curly braces hold key value pairs.Hashes are similiar to array actually another name for hashes are associative arrays.I learned 4 things you can do with a hash is Insert which adds  a key and an value, Remove which deletes a key and its associated value, Search which is using a key to find its value and Modify which is switching the value of the key value pair. Using hashes in ruby was fairly easly creating the text based game because when my group wanted to switcharound choices for the game to we just needed to reassign the value.
```
Comment (SunkissedQueen) [inline interview-questions.md:13]: wonderful

File: ruby-challenges.rb [hunk 13]
```diff
@@ -10,7 +10,10 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
-
+def singleL(array, letter)
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:13]: nice

File: ruby-challenges.rb [hunk 14]
```diff
@@ -19,11 +22,28 @@
 nums_array2 = [25, 17, 47, 11]
 # Expected output: 100
 
-
+def sumNum(array)
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:25]: yes let sum do that lifting

File: ruby-challenges.rb [hunk 15]
```diff
@@ -33,3 +53,48 @@
 # Expected output example: my_bike.pedal_faster(18) => 28
 # Expected output example: my_bike.brake(5) => 23
 # Expected output example: my_bike.brake(25) => 0
+
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:56]: quality work

**Review by SunkissedQueen — APPROVED** [review]
Davon, you did some good work. I see many improvements. Especially those interview answers. Keep that approach. For the code challenge, practice breaking it down into small chunks. First build the function, 
`const functionName = (parameter) => {return parameter}`
then introduce one method at a time to the return statement. Taking your time in the beginning saves a lot of time on the other end.

File: code-challenges.test.js (hunk 4 above)
Comment (DavonKenn) [reply code-challenges.test.js:78]: NO it doesnt

File: code-challenges.test.js (hunk 5 above)
Comment (DavonKenn) [reply code-challenges.test.js:96]: I wasnt paying attention

**Pull request merged**